### ISSUE TITLE
Supported input types for tf.nn.quantized_conv2d

### ISSUE THREAD

**Issue by rvinas** [tensorflow/tensorflow#23157]
**System information**
- TensorFlow version: 1.11.0
- Doc Link: https://www.tensorflow.org/api_docs/python/tf/nn/quantized_conv2d
- Have I written custom code: Check [this question](https://stackoverflow.com/questions/52922980/error-when-using-quantized-conv2d-with-tf-qint8-inputs).
- OS Platform and Distribution: macOS Mojave
- TensorFlow installed from: pip binary
- Bazel version: N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A
- Exact command to reproduce: N/A
- Mobile device: N/A

**Describe the documentation issue**

The documentation for [tf.nn.quantized_conv2d](https://www.tensorflow.org/api_docs/python/tf/nn/quantized_conv2d) says that `input` and `filter` must be one of the following types: `qint8`, `quint8`, `qint32`, `qint16`, `quint16`.

However, the only registered type for these arguments is `tf.quint8`. See also [this question](https://stackoverflow.com/questions/52922980/error-when-using-quantized-conv2d-with-tf-qint8-inputs).

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by wt-huang** [comment]
@rvinas It is true that tf.quint8 is the only registered type for input and filter as it covers most cases in conv2d.
We will certainly look into other types as well.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @wt-huang: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by shoubhik** [comment]
@wt-huang, if I want to create custom op for `tf.nn.quantized_conv2d` where I register the data types I want, is there a way to do so? What I am thinking is registering my own kernels only like this

```
REGISTER_KERNEL_BUILDER(
    Name("QuantizedConv2D")
        .Device(DEVICE_CPU)
        .TypeConstraint<quint8>("Tinput")
        .TypeConstraint<quint8>("Tfilter")
        .TypeConstraint<qint32>("out_type"),
    QuantizedConv2DOp< <qint16, qint16, qint32, Im2ColConvFunctor>);
```
